web-ng/src/app/services/router/router.service.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ActivatedRoute } from '@angular/router';
+import { HttpParams } from '@angular/common/http';
+
+/**
+ * Exposes application state in the URL as streams to other services
+ * and components, and provides methods for altering said state.
+ */
+@Injectable({
+  providedIn: 'root',
```

Comment:
Can this be injected to MainPageModule?

Reply:
I think this is probably a good idea, but at present all of our services are provided in root instead of as service modules. According to https://angular.io/guide/module-types, they should only be imported into App Module. Let's follow up on this in #349?